proto/regen/ecocredit/v1beta1/state.proto

```diff
@@ -0,0 +1,208 @@
+    // quantified and verified.
+    google.protobuf.Timestamp end_date = 7 [ (gogoproto.stdtime) = true ];
+}
+
+// CreditType defines the measurement unit/precision of a certain credit type
+// (e.g. carbon, biodiversity...)
+message CreditType {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 5,
+        primary_key: {fields: "abbreviation"}
+    };
+
+    // the type of credit (e.g. carbon, biodiversity, etc)
+    string name = 1;
+
+    // abbreviation is a 1-3 character uppercase abbreviation of the CreditType
+    // name, used in batch denominations within the CreditType. It must be unique.
+    string abbreviation = 2;
+
+    // the measurement unit (e.g. kg, ton, etc)
+    string unit = 3;
+
+    // the decimal precision
+    uint32 precision = 4;
+}
+
+// CreditTypeSequence stores and increments the sequence number for credit types
+message CreditTypeSequence {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 6,
+        primary_key: {fields: "abbreviation"}
+    };
+
+    string abbreviation = 1;
+
+    uint64 id = 2;
+}
+
+// ProjectSequence stores and increments the sequence number for projects
+message ProjectSequence {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 7,
+        primary_key: {fields: "project_id"}
+    };
+
+    // project_id is the id of the project
+    string project_id = 1;
+
+    // next_id is the sequence number for the project
+    uint64 next_id = 2;
+}
+
+// EcocreditBalance stores each users credit balance
+message EcocreditBalance {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 8,
+        primary_key: {fields: "address,batch_id"}
+    };
+
+    // address is the address of the credit holder
+    string address = 1;
+
+    // batch_id is the id of the credit batch
+    uint64 batch_id = 2;
+
+    // tradable is the tradable amount of credits
+    string tradable = 3;
+
+    // retired is the retired amount of credits
+    string retired = 4;
+}
+
+// BatchSequence tracks the sequence number for batches within a class
+message BatchSequence {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 10,
+        primary_key: {fields: "class_id"}
+    };
+
+    // class_id is the id of the class for a batch
+    uint64 class_id = 1;
+
+    // num_batches is a sequence number incrementing on each issued batch
+    uint64 num_batches = 2;
+}
+
+// BatchSupply tracks the supply of a credit batch
+message BatchSupply {
+    option (cosmos.orm.v1alpha1.table) = {
+        id: 11,
+        primary_key: {fields: "id"}
+    };
+
+    // id is the id of the batch
+    uint64 id = 1;
+
+    // total_amount is the total number of active credits in the credit batch.
+    // Some of the issued credits may be cancelled and will be removed from
+    // total_amount and tracked in amount_cancelled. total_amount and
+    // amount_cancelled will always sum to the original credit issuance amount.
+    string total_amount = 2;
```

Comment:
Where are we tracking retired amount? That's different from cancelled

Reply:
it wasn't tracked in the original design. i can split the supply into tradable and retired

Reply:
https://github.com/regen-network/regen-ledger/pull/700/commits/40bbe1ed417b578073d2f232230562cd54a61cf5